### ISSUE TITLE
No Analyzers Found

### ISSUE THREAD

**Issue by hackdefendr** [TheHive-Project/Cortex#88]
# No Analyzers Found

### Request Type
Issue

### Work Environment

| Question              | Answer
|---------------------------|--------------------
| OS version (server)       | Ubuntu 16.04
| OS version (client)       | Ubuntu 16.04
| Cortex version / git hash   | 2.0.3
| Package Type              | Binary
| Browser type & version    | Chrome

### Problem Description
When I click on the Analyzers link, there is nothing showing up. The log shows that all of the analyzers loaded, but for some reason nothing is showing up in the Web interface.

### Steps to Reproduce
1. Install and Configure Cortex
1. Pull the Cortex-Analyzers repository
1. Configure Cortex to read from the folder where the analyzers are located.
1. Launch a browser and log into Cortex
1. Click on Analyzers

### Possible Solutions
I have looked in the logs, but nothing is showing me what the problem is.

### Complementary information
Here is my config from application.conf:

```
## ANALYZERS
#
analyzer {
  # Absolute path where you have pulled the Cortex-Analyzers repository.
  path = ["/usr/share/Cortex-Analyzers/analyzers/"]

  # Sane defaults. Do not change unless you know what you are doing.
  fork-join-executor {

    # Min number of threads available for analysis.
    parallelism-min = 2

    # Parallelism (threads) ... ceil(available processors * factor).
    parallelism-factor = 2.0

    # Max number of threads available for analysis.
    parallelism-max = 4
  }
}
```

**Comment by nadouani** [comment]
Well, as stated on the Quick guide, you need to login to Cortex as a `OrgAdmin`, and configure/enable the analyzers you want.

By default, all analyzers are disabled.

**Comment by nadouani** [comment]
Did you take a look to Cortex 2 official announcement blog post ?

https://blog.thehive-project.org/2018/03/29/unveiling-cortex-2/

**Comment by hackdefendr** [comment]
I am logged in a OrgAdmin and I see no Analyzers to enable or configure. Nothing is listed at all.

What on that blog page am I supposed to key in on?  Skimming through it all I see is the explanation of the new roles added in Cortex 2.  Nothing about HOW to enable analyzers.

**Comment by hackdefendr** [comment]
Never mind.

**Comment by nadouani** [comment]
If you see `no analyzers to enable or configure`that means that you cortex is not finding any analyzer definition in the `analyzer.path` location that you set on your `application.conf` file.

So, again, Cortex on startup, searched for analyzer definition using the `analyzer.path` config, and then lists them on the UI, and allo OrgAdmins to enable the ones he wants.

**Comment by nadouani** [comment]
OK, never mind

**Comment by hudsonmanieri** [comment]
I got the same problem on CENTOS 7.
I just changed the directory permissions such as: 

chown -R cortex:root Cortex-Analyzers
chmod -R 770 Cortex-Analyzers

Worked like a charm

**Comment by liamcs98** [comment]
For you future people
/opt/cortex-2.0.4/conf/application.conf 
(or where ever you have you conf file)

Long story but the path might or might not need quotes.
```
analyzer {
  # Directory that holds analyzers
  path = [
	/opt/Cortex-Analyzers/analyzers/
  ]

  fork-join-executor {
    # Min number of threads available for analyze
    parallelism-min = 2
    # Parallelism (threads) ... ceil(available processors * factor)
    parallelism-factor = 2.0
    # Max number of threads available for analyze
    parallelism-max = 4
  }
}
```

**Comment by DevTeam34** [comment]
Can someone assist with this issue to. I ran all the steps according to the instructions but when log into the cortex as orgadmin i do not see the analyzers. I didnt make any changes to the file path so assuming everything is by default I would of thought the default file path to the analyzers was "/opt/Cortex-Analyzers/analyzers" but that didnt work.

I then did the troubleshooting as suggested by the community which consisted of checking the permissions and trying with and without quotes and still came up short. I then went to check the file path with the ls command and realized that path doesn't exist so I figured the i try and reinstall the cortex analyzers  that states: "destination path 'Cortex-Analyzers' already exists and is not an empty directory." As the processed continued, I saw that items were getting installed to the /usr/lib/python3/dist-packages file path. So I went to check the files and saw alot of items pertaining the analyzers, so I used that path hoping that would of been the fix and still came up short.

So with my issues the questions I have for the community are the following:
1. When I git clone the Cortex-Analyzers/analyzers (git clone https://github.com/TheHive-Project/Cortex-Analyzer) what is the file path that these items are being stored to?
2. What is the actual default file path to the cortex analyzers?
3. In the following command: ( for I in $(find Cortex-Analyzers -name 'requirements.txt'); do sudo -H pip2 install -r $I; done && \
> for I in $(find Cortex-Analyzers -name 'requirements.txt'); do sudo -H pip3 install -r $I || true; done)
Where is the variable Cortex-Analyzers being pulled from? I cant seem to find the location?

Thanks in advance for the help!
